lib/snowplow-tracker/emitters.rb

```diff
@@ -234,8 +372,65 @@ def good_status_code?(status_code)
             :http_post
   end
 
+  # This {Emitter} subclass provides asynchronous event sending. Whenever the
+  # buffer is flushed, the AsyncEmitter places the flushed events in a work
+  # queue. The AsyncEmitter asynchronously sends events in this queue using a
+  # thread pool of a fixed size. The size of the thread pool is 1 by default,
+  # but can be configured as part of the options hash during initialization.
+  #
+  # @see Emitter
+  # @api public
   class AsyncEmitter < Emitter
     Contract KeywordArgs[endpoint: String, options: Optional[STRICT_CONFIG_HASH]] => Any
+    # Create a new AsyncEmitter object. The endpoint is required.
+    #
+    # @example Initializing an AsyncEmitter with all the possible extra configuration.
+    #   success_callback = ->(success_count) { puts "#{success_count} events sent successfully" }
+    #   failure_callback = ->(success_count, failures) do
+    #     puts "#{success_count} events sent successfully, #{failures.size} sent unsuccessfully"
+    #   end
+    #
+    #   Emitter.new(endpoint: 'd3rkrsqld9gmqf.cloudfront.net',
```

Comment:
I'd potentially remove the old cloudfront url examples. The Snowplow Cloudfront Collector is an old piece of technology that is long deprecated. I'd suggest altering it to something like `collector.example.com`.